package/MDAnalysis/analysis/pcz.py

```diff
@@ -0,0 +1,481 @@
+        #         first least-squares fitted to the mean structure, then 
+        #         the mean structure is subtracted from it. The covariance
+        #         matrix is built up in as big blocks as memory will permit.
+        covar = np.zeros((3*self._f.natoms, 3*self._f.natoms))
+        blocksize = self.numframes * 2
+        toobig = True
+        while toobig:
+            blocksize = blocksize/2
+            try:
+                trj = np.zeros((blocksize, self.natoms * 3))
+                toobig = False
+            except MemoryError:
+                blocksize = blocksize/2
+                
+        num_full_blocks = self.numframes/blocksize
+        remainder = self.numframes%blocksize
+
+        frames_so_far = 0
+        for i in range(num_full_blocks):
+            for j in range(blocksize):
+                mob = self._f.coords(j + frames_so_far, reference_structure = self.avg) - self.avg
+                trj[j,:] = mob.flatten()
+            covar += np.dot(trj.T, trj.conj())
+            frames_so_far += blocksize
+
+        trj = np.zeros((remainder, self.natoms * 3))
+        for j in range(remainder):
+            mob = self._f.coords(j + frames_so_far, reference_structure = self.avg) - self.avg
+            trj[j,:] = mob.flatten()
+        covar += np.dot(trj.T, trj.conj())
+        covar = covar/self.numframes
+
+        # Step 4: Diagonalise the covariance matrix; reorder the eigenvectors
+        #         and eigenvalues into descending importance; reduce to the
+        #         number required to explain the desired fraction of the total
+        #         variance (if specified).
+        self.evals, self.evecs = eigh(covar)
+        self.evals = self.evals[-1::-1]
+        self.evecs = self.evecs[:, -1::-1].T
+        self.nvecs = len(self.evals)
+
+        cs = np.cumsum(self.evals)
+        cs = cs/cs[-1]
+
+        if captured_variance is not None:
+            if captured_variance < 1.0:
+                i = 0
+                while cs[i] < captured_variance:
+                    i += 1
+                self.nvecs = i +1 
+            else:
+                self.nvecs = int(captured_variance)
+
+            self.evals = self.evals[:self.nvecs]
+            self.evecs = self.evecs[:self.nvecs]
+
+        self.captured_variance = cs[self.nvecs - 1]
+
+        # Step 5: If only a subset of the 3N eigenvectors have been
+        #         saved, precalculate and store the projection data.
+        if self.captured_variance < 1.0:
+            self._proj = np.zeros((self.nvecs, self.numframes))
+            for i in range(self.numframes):
+                x = self._f.coords(i, reference_structure = self.avg) - self.avg
+                self._proj[:,i] = np.dot(x.flatten(), self.evecs.T)
+            
+            
+    def proj(self, ev):
+        """
+        Returns the projection of each frame in the trajectory along
+        eigenvector *ev* (zero-based), as a numpy array.
+
+        :Arguments:
```

Comment:
That doesn't solve the problem. For example if you simulate a channel in the closed state and open state you will likely have two seperate simulations (or one very very long simulation to see a transition). Since you have two simulations you would usually load two universes in MDA. Currently I can't do a PCA on the open state and look at the projection in the closed state.